addons/website/models/website_page.py

```diff
@@ -199,6 +196,16 @@ def _handle_url_update(self, page, website_id, url, lang, redirect_old_url=None,
                 website._handle_homepage_url(page_url_normalized)
                 if website.homepage_url == page_url_normalized['homepage_url']:
                     website.homepage_url = url
+        if redirect_old_url:
```

Comment:
This has been moved out of the `if page.url != url` to handle the case of a redirection when the website default language is different of the user default language. In this case, the redirection is done thanks to a translation and the code do not enter in the `if page.url != url` condition.